app/presenters/date_question_presenter.rb

```diff
@@ -12,22 +12,18 @@ def response_label(value)
     end
   end
 
-  def selected_day
-    return unless response_for_current_question.is_a? Hash
-
-    response_for_current_question[:day]
-  end
-
-  def selected_month
-    return unless response_for_current_question.is_a? Hash
-
-    response_for_current_question[:month]
-  end
-
-  def selected_year
-    return unless response_for_current_question.is_a? Hash
-
-    response_for_current_question[:year]
+  def parsed_response
+    return {} if current_response.blank?
+    return current_response if current_response.is_a?(Hash)
+
+    date = @node.parse_input(current_response)
+    {
+      day: date.day,
+      month: date.month,
+      year: date.year,
+    }
```

Comment:
How come we don't just pass current response to `@node.parse_input` and return the underlying Date object? I.e. why do we need to make another `Hash`?

Reply:
We don't know if a user has inputted a full date to allow us to create one.

For example, if a user just enters the month and year field of a date.